core/controllers/collection_editor.py

```diff
@@ -220,6 +267,27 @@ def put(self, collection_id):
                 raise self.InvalidInputException(
                     'Cannot unpublish a collection.')
 
+        elif is_publicized is not None:
+            if is_publicized:
+                try:
+                    collection.validate(strict=True)
+                except utils.ValidationError as e:
+                    raise self.InvalidInputException(e)
+
+                rights_manager.publicize_exploration(
```

Comment:
I'm pretty sure publicize_exploration isn't the right method to call here -- please be careful; this sort of thing is important to get right. Ditto below.